misk/src/main/kotlin/misk/concurrent/RealExecutorServiceFactory.kt

```diff
@@ -63,28 +63,44 @@ internal class RealExecutorServiceFactory @Inject constructor(
   }
 
   override fun single(nameFormat: String): ExecutorService {
+    check(!nameFormat.contains("%d")) {
+      "thread index %d placeholder unexpected: $nameFormat"
+    }
     checkCreate()
+
     val threadFactory = threadFactory(nameFormat)
     return Executors.newSingleThreadExecutor(threadFactory)
         .also { executors[nameFormat] = it }
   }
 
   override fun fixed(nameFormat: String, threadCount: Int): ExecutorService {
     checkCreate()
+    check(threadCount == 1 || nameFormat.contains("%d")) {
```

Comment:
This seems wrong? Isn't the second one ! contains `%d`

Maybe split this out.. 
= 1 and contains

or 
> 1 and not contains?

Reply:
the `1` here could be dynamic, like coming from a config file or a dynamic flag. I don’t want to require you to remove the `%d` if you change a config option from 2 to 1.

But I also don’t want to explode if you specify `1` without `%d`, because it’s possible that it _isn’t_ coming from a config file.